src/api/games/game-projects.controller.ts

```diff
@@ -253,14 +248,9 @@ export class GameProjectsController {
     @Param('id') id: number,
     @CurrentUser() user: UserJWTPayload,
   ) {
-    const target = await this.gamesService.findOne(id);
-
-    if (user.username !== target.username) {
-      throw new ForbiddenException(
-        "You don't have permission to delete the file of this project",
-      );
-    }
+    await this.gamesService.verifyOwner(id, user);
 
+    const target = await this.gamesService.findOne(id);
```

Comment:
extract a method of gamesService

Reply:
Yeah I've implemented a new service called `game.logic.service.ts`, check out my new updates.